tools/universe/package_publisher.py

```diff
@@ -0,0 +1,163 @@
+                with open(last_filename, 'r') as last_file, open(this_filename, 'r') as this_file:
+                    filediff = ''.join(difflib.unified_diff(
+                        last_file.readlines(), this_file.readlines(),
+                        fromfile='{}/{}'.format(lastnum, filename),
+                        tofile='{}/{}'.format(lastnum + 1, filename)))
+                    if filediff:
+                        filediffs[filename] = filediff
+        else:
+            filediffs = {}
+            removed_files = {}
+            added_files = os.listdir(this_dir)
+
+        # create a user-friendly diff for use in the commit message:
+        resultlines = [
+            'Changes since revision {}:\n'.format(lastnum),
+            '{} files added: [{}]\n'.format(len(added_files), ', '.join(added_files)),
+            '{} files removed: [{}]\n'.format(len(removed_files), ', '.join(removed_files)),
+            '{} files changed:\n\n'.format(len(filediffs))]
+        if self._commit_desc:
+            resultlines.insert(0, 'Description:\n{}\n\n'.format(self._commit_desc))
+        # surround diff description with quotes to ensure formatting is preserved:
+        resultlines.append('```\n')
+        filediff_names = list(filediffs.keys())
+        filediff_names.sort()
+        for filename in filediff_names:
+            resultlines.append(filediffs[filename])
+        resultlines.append('```\n')
+        commitmsg_path = os.path.join(scratchdir, 'commitmsg.txt')
+        with open(commitmsg_path, 'w') as commitmsg_file:
+            commitmsg_file.write(self._pr_title)
+            commitmsg_file.writelines(resultlines)
+        # commit the change and push the branch:
+        cmds = ['cd {}'.format(os.path.join(scratchdir, 'universe')),
+                'git add .',
+                'git commit -q -F {}'.format(commitmsg_path)]
+        if self._dry_run:
+            # ensure the debug goes to stderr...:
+            cmds.append('git show -q HEAD 1>&2')
+        else:
+            cmds.append('git push origin {}'.format(branch))
+        ret = os.system(' && '.join(cmds))
+        if ret != 0:
+            raise Exception('Failed to push git branch {} to Universe.'.format(branch))
+        return (branch, commitmsg_path)
+
+
+    def _create_universe_pr(self, branch, commitmsg_path):
+        if self._dry_run:
+            log.info('[DRY RUN] Skipping creation of PR against branch {}'.format(branch))
+            return None
+        headers = {
+            'User-Agent': 'release_builder.py',
+            'Content-Type': 'application/json',
+            'Authorization': 'Basic {}'.format(self._enc_github_token)}
+        with open(commitmsg_path) as commitmsg_file:
+            payload = {
+                'title': self._pr_title,
+                'head': branch,
+                'base': self._release_branch,
+                'body': commitmsg_file.read()}
+        conn = http.client.HTTPSConnection('api.github.com')
```

Comment:
We earlier used `urllib` and here we use `http`. Could we not use `requests` for both cases instead?